### PR TITLE
feat(swapper): display buyTxid on last step

### PR DESCRIPTION
**Pull request by gomesalexandre**
THOR is back up, all 🛍️ here
~~⚠️ Can't be tested with THOR while THOR swaps are down, but this can be tested with Li.Fi which is also cross-chain~~

## Description

Now that we have the concept of hops and of a last step, we should be able to display the buy Txid holistically (not only for THOR, which is the issue this PR originally tackles).

Note, we currently represent THOR swaps as a single step, though this isn't an issue in the current exchange swapper implementation.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes the very old https://github.com/shapeshift/web/issues/2797 - eventually, when we have *actual* multi-hop UI, we will be able to do things better instead of replacing the tx link

## Risk

Tx link may be borked, test accordingly

## Testing

- When the initiating chain Tx is still ongoing for a THOR/Li.Fi cross-chain Tx, the initiating chain Tx
  is displayed
- As soon as the destination chain Tx message arrives (which is the exact same moment we
  consider a THOR Tx as completed, but may not be the case for Li.Fi as we can have a buy-side Txid while confirmation is still pending), the Tx link should change to either the destination chain one for Li.Fi, or the viewblock link for THOR
- There are no regressions with any other swappers Tx link, both same chain and cross-chain

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

- Li.Fi

image
image

- THOR
Screenshot 2023-07-31 at 16 01 33
Screenshot 2023-07-31 at 16 02 07
Screenshot 2023-07-31 at 16 02 35
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4970** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4970" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4970?utm_source=stack-comment).

File: src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx [hunk 1]
```diff
@@ -140,19 +142,39 @@ export const TradeConfirm = () => {
...
+      }),
+    [lastStep?.buyAsset.explorerTxLink, tradeQuoteStep?.sources],
+  )
+
+  const txLink = useMemo(() => {
+    if (buyTxHash) return getBuyTxLink(buyTxHash)
+    if (sellTxHash) return getSellTxLink(sellTxHash)
+  }, [buyTxHash, getBuyTxLink, getSellTxLink, sellTxHash])
+
+  console.log({ sellTxHash, buyTxHash, txLink })
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx:176]: For testing purposes only, remove me when opening

File: src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx [hunk 2]
```diff
@@ -140,19 +142,39 @@ export const TradeConfirm = () => {
...
-        tradeId: sellTxId,
+        tradeId: sellTxHash,
       }),
     [tradeQuoteStep?.sellAsset.explorerTxLink, tradeQuoteStep?.sources],
   )
 
+  const getBuyTxLink = useCallback(
+    (buyTxHash: string) =>
+      getTxLink({
+        name: tradeQuoteStep?.sources[0]?.name,
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx:164]: `lastStep`

File: src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx [hunk 3]
```diff
@@ -381,13 +403,13 @@ export const TradeConfirm = () => {
               {tradeWarning}
               {sendReceiveSummary}
               <Stack spacing={4}>
-                {sellTxHash && (
+                {txHash && (
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx:406]:
```suggestion
                {txLink && (
```

File: src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx]: 5aaa4ce85

File: src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx (hunk 3 above)
Comment (gomesalexandre) [reply src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx]: da219138a6b98aafbdca6fe84ac9362ae54af163

File: src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx [hunk 4]
```diff
@@ -119,6 +120,7 @@ export const TradeConfirm = () => {
   }, [dispatch, tradeQuote])
 
   const tradeQuoteStep = useAppSelector(selectFirstHop)
```
Comment (0xApotheosis) [inline src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx:122]: We should probably rename this to `firstStep` now.

**Review by 0xApotheosis — APPROVED** [review]
Beautiful!

**Pull request merged**